tests/integration-tests/tests/schedulers/test_torque.py

```diff
@@ -143,6 +146,47 @@ def _test_job_arrays_and_parallel_jobs(remote_command_executor, region, stack_na
     _assert_job_completed(remote_command_executor, parallel_job_id)
 
 
+def _test_jobs_executed_concurrently(remote_command_executor, max_slots):
+    logging.info("Testing jobs are executed concurrently and nodes are fully allocated")
+    torque_commands = TorqueCommands(remote_command_executor)
+
+    # GIVEN: a cluster with 3 free nodes
+    assert_that(torque_commands.compute_nodes_count()).is_equal_to(0)
+    result = torque_commands.submit_command("sleep 1", nodes=3)
+    job_id = torque_commands.assert_job_submitted(result.stdout)
+    torque_commands.wait_job_completed(job_id)
+    assert_that(torque_commands.compute_nodes_count()).is_equal_to(3)
+
+    # WHEN: an array job that requires 3 nodes and all slots is submitted
+    jobs_start_time = int(remote_command_executor.run_remote_command("date +%s").stdout)
+    job_exec_time = 30
+    job_ids = []
+    for i in range(0, 3 * max_slots):
+        result = torque_commands.submit_command(
+            f"sleep {job_exec_time} && hostname > /shared/job{i} && date +%s >> /shared/end_time", nodes=1, slots=1
+        )
+        job_id = torque_commands.assert_job_submitted(result.stdout)
+        job_ids.append(job_id)
+
+    # THEN: cluster scales down correctly after completion
+    watch_compute_nodes(torque_commands, minutes(10), 0)
+    for id in job_ids:
+        _assert_job_completed(remote_command_executor, id)
+
+    # THEN: each host executes 4 jobs in the expected time
+    jobs_to_hosts_count = (
+        remote_command_executor.run_remote_command("cat /shared/job* | sort | uniq -c | awk '{print $1}'")
+        .stdout.strip()
+        .splitlines()
+    )
+    assert_that(jobs_to_hosts_count).is_equal_to(["4", "4", "4"])
+    # verify execution time
+    jobs_completion_time = int(
+        remote_command_executor.run_remote_command("cat /shared/end_time | sort -n | tail -1").stdout.split()[-1]
```

Comment:
Is there an easier way to do this? Perhaps through a `@retry` decorator.

Reply:
how would you do it with a retry? just bear in mind that we want to retrieve the timestamp from the host and not from the jenkins instance since there might be some misalignment